File: GiaThietABach.cpp
#include <iostream>
using namespace std;
bool checkSoNT(int n)
{
	if (n < 2)
        return false;
    int count = 0;
    for (int i = 1; i <= n; i++)
	{
		if (n % i == 0)
            count += 1;
	} 
    if (count == 2)
        return true;
    else
    	return false;
}
int giaThiet(int n)
{
	int count = 0;
	 for (int i = 2; i <= n; i++)
	 {
	 	if (checkSoNT(i))
	 	{
	 		//cout << i << endl;
	 		for (int j = 2; j <= n; j++)
		    {
		    	if (checkSoNT(j))
		        {
		        	for (int o = j+1; o <= n; o++) //o in range(j+1, n+1):
		            {
		            	int tong = 0;
	                    if (checkSoNT(o))
	                    {
							tong = j + o + 1;
	                        int dem = 0;
	                        for (int l = j+1; l < o; l++) //l in range(j+1, o):
	                            if (checkSoNT(l))
	                                dem+=1;
	                        if (tong == i && dem == 0)
	                        {
	                        	//cout << checkSoNT(tong) << endl;
	                        	count +=1;
	                        	//cout << j << "+" <<  o <<  "=" <<  tong <<  "i =" << i << endl;
							}    
						}    
					}       
				}       
		    } 
		}
		 	      
	}        
    return count;
}
int main()
{
	int t; 
	cin >> t; //= int(input())
	for(int i = 0; i < t; i++)
	{
		int n; // = int(input())
	    cin >> n;
		int k; 
		cin >> k; // = int(input())
	    if (giaThiet(n) >= k)
	        printf("YES\n");
	    else
	        printf("NO\n");
	}	        
}     

File: GiaThietABach.py

def checkSoNT(n):
    if n < 2:
        return False
    count = 0
    for i in range(1, n+1):
        if n % i == 0:
            count += 1
    if count == 2:
        return True
def giaThiet(n):
    count = 0
    for i in range(2, n+1):
            for j in range(2, n):
                if checkSoNT(j):
                    for o in range(j+1, n+1):
                        tong = 0
                        if checkSoNT(o):
                            tong = j + o + 1
                            dem = 0
                            for l in range(j+1, o):
                                if checkSoNT(l):
                                    dem+=1
                            if tong == i and dem == 0 and checkSoNT(tong):    
                                count +=1
                                #print(j, "+", o, "=", tong, "i =", i)
    return count
t = int(input())
for i in range(t):
    n = int(input())
    k = int(input())
    if giaThiet(n) >= k:
        print("YES")
    else:
        print("NO")    